eng/pipelines/checkout-task.yml

```diff
@@ -1,5 +1,15 @@
-# Shallow checkout sources on Windows
+# Shallow checkout sources
 steps:
-  - checkout: self
-    clean: true
-    fetchDepth: 1 # Fetch only one commit
\ No newline at end of file
+  - checkout: none
+  - script: |
```

Comment:
This reduces checkout time on windows from 2 minutes to 30 seconds

Reply:
A similar change is being flowed from our servicing branches only it uses the Agent's git copy instead of relying on git being installed on the system. https://github.com/dotnet/roslyn/pull/62061/files

Reply:
alright, I'll submit a PR to main with just this change